### ISSUE TITLE
Add ability to set URL for ngsw.json at runtime

### ISSUE THREAD

**Issue by anhallbe** [angular/angular#51457]
### Which @angular/* package(s) are relevant/related to the feature request?

service-worker

### Description

TL;DR: We cannot serve `ngsw-worker.js` and `ngsw.json` from the same origin, so we'd like a way to configure where `ngsw.json` is fetched from.

Our setup requires us to serve the application (i.e files produced by `ng build`) from an external CDN. The CDN URL and path cannot be known at build-time, and is set by the server when the index file is rendered (e.g with [Thymeleaf](https://www.thymeleaf.org)). Think:

```html
// index.html
<script>
  window.APP_CONFIG = {
    cdnUrl: [[${CDN_URL}]]
  };
</script>

<script th:src="@{${CDN_URL}+'polyfills.js'}"></script>
<script th:src="@{${CDN_URL}+'runtime.js'}"></script>
<script th:src="@{${CDN_URL}+'main.js'}"></script>
```

When trying to add  `@angular/service-worker` for offline support into the mix, we run into the issue that the `ngsw-worker.js` script cannot be served from the CDN like other files, because of browser restrictions.

We *can* work around this by having the `ngsw-worker.js` be served from the page origin. However, we'd still like for the `ngsw.json` manifest to be served together with the rest of the app from the CDN. And this is the issue. There is no way to tell the service worker to fetch the manifest from a different path/url.

### Proposed solution

Add a *runtime* config property that allows to override the default `ngsw.json` path. For example: 

```typescript
const CDN_URL = "https://the-cdn/versioned/path/to/app"; // Set by the server at runtime

ServiceWorkerModule.register("ngsw-worker.js", {
	// ...
	manifestUrl: `${CDN_URL}/ngsw.json`,
}),
```

Note that for our setup, the CDN URL is not known at build-time, so it cannot be a part of `ngsw-config.json`.

### Alternatives considered

One alternative we tried was to serve `ngsw.json` from the page origin (i.e the same as `ngsw-worker.json`).

However this is not ideal, as the CDN files are deployed more or less independently from the index page.

**Comment by JeanMeche** [comment]
Per https://github.com/angular/angular/pull/63364, we won't be addressing this issue.

The message is also visible here: https://angular.dev/ecosystem/service-workers
